packages/source-iottwinmaker/src/scene-module/SceneMetadataModule.spec.ts

```diff
@@ -0,0 +1,132 @@
+import { IoTTwinMakerClient } from '@aws-sdk/client-iottwinmaker';
+import { SceneMetadataModule } from './SceneMetadataModule';
+import { SecretsManagerClient } from '@aws-sdk/client-secrets-manager';
+
+/*eslint-disable @typescript-eslint/no-explicit-any */
+
+class MockResponse {
+  data: any;
+
+  constructor(data: any) {
+    this.data = data;
+  }
+
+  arrayBuffer = jest.fn().mockImplementation(() => this.data);
+}
+
+global.Response = MockResponse as any;
+
+const sceneMetadataModule = new SceneMetadataModule({
+  workspaceId: 'ws-id',
+  sceneId: 'scene-id',
+  twinMakerClient: new IoTTwinMakerClient({}),
+  secretsManagerClient: new SecretsManagerClient({}),
+});
+
+describe('getSceneInfo', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should get a valid scene info', async () => {
+    const sendSpy = jest.spyOn(IoTTwinMakerClient.prototype, 'send').mockImplementation(() => {
+      return Promise.resolve({
+        contentLocation: 'mock-contentLocation',
+        s3Location: 'mock-s3Location',
+        capabilities: ['MATTERPORT'],
+      });
+    });
+
+    const expected = {
+      contentLocation: 'mock-contentLocation',
+      s3Location: 'mock-s3Location',
+      capabilities: ['MATTERPORT'],
+    };
+    const result = await sceneMetadataModule.getSceneInfo();
+
+    expect(sendSpy).toBeCalledTimes(1);
+    expect(result).toEqual(expected);
+  });
+
+  it('should get error when API failed', async () => {
+    const sendSpy = jest.spyOn(IoTTwinMakerClient.prototype, 'send').mockImplementation(() => {
+      throw 'TwinMaker API failed';
+    });
+
+    let error: any;
+    try {
+      await sceneMetadataModule.getSceneInfo();
+    } catch (e) {
+      error = e;
+    }
+
+    expect(sendSpy).toBeCalled();
+    expect(error).toEqual('TwinMaker API failed');
+  });
+});
+
+describe('updateScene', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should update the scene', async () => {
+    const sendSpy = jest.spyOn(IoTTwinMakerClient.prototype, 'send').mockImplementation(() => {
+      return Promise.resolve();
+    });
+
+    await sceneMetadataModule.updateScene(['MATTERPORT'], { MATTERPORT_SECRET_ARN: 'dummySecretARN' });
+
+    expect(sendSpy).toBeCalledTimes(1);
+  });
+
+  it('should get error when API failed', async () => {
+    const sendSpy = jest.spyOn(IoTTwinMakerClient.prototype, 'send').mockImplementation(() => {
+      throw 'TwinMaker API failed';
+    });
+
+    let error: any;
```

Comment:
no any

Reply:
Not required after refactoring - Removed the code